app/models/tag.rb

```diff
@@ -23,6 +23,68 @@ class Tag < ActiveRecord::Base
   # the order is important, and it is the order in which they appear in the tag wrangling interface
   USER_DEFINED = ['Fandom', 'Character', 'Relationship', 'Freeform']
 
+  def is_this_a_large_tag(length_of_time_in_cache,how_long_it_takes_to_compute_size)
+    if length_of_time_in_cache < 40 || how_long_it_takes_to_compute_size < ( ArchiveConfig.TAGINGS_COUNT_DEFAULT_FOR_LARGE_TAGS || 4 )
+      if self.large_tag
+        self.large_tag = false
+        self.save
+        return
+      end
+    end
+    return if self.large_tag
+    self.large_tag = true
+    self.save
+  end
+
+  def taggings_count_expiry(count)
+    expiry_time = count/(ArchiveConfig.TAGINGS_COUNT_CACHE_DIVISOR || 2000 )
+    [[expiry_time, (ArchiveConfig.TAGINGS_COUNT_MIN_TIME||3)].min, (ArchiveConfig.TAGINGS_COUNT_MIN_TIME||60)].max
```

Comment:
Pretty much the same as above! "Taggings" should have two Gs and there should probably be a comment somewhere explaining why the fallback values are 3 and 60 here.